### PR TITLE
feat(web-components): add set focus methods

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Add set focus methods to remaining interactive components: breadcrumb, card, checkbox, chip, link, radio, switch and tab

## Related issue
#372 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Comment by ad9242** [comment]
We should add to the unit tests to ensure code coverage doesn't drop

File: packages/web-components/src/components/ic-card/ic-card.spec.ts [hunk 1]
```diff
@@ -183,4 +183,26 @@ describe("ic-card", () => {
...
+      components: [Card],
+      html: `<ic-card heading="Card" message="This is a clickable card rendered as a button" clickable=true></ic-card>`,
+    });
+
+    await page.rootInstance.focus;
+
+    await page.rootInstance.setFocus().toHaveBeenCalled;
+  });
+
+  it("should call 'setFocus' when card as a button is focused", async () => {
```
Comment (ad9242) [inline packages/web-components/src/components/ic-card/ic-card.spec.ts:198]: this seems very similar to the test above apart from href prop. Should the description be changed?

File: packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts [hunk 2]
```diff
@@ -119,4 +119,15 @@ describe("ic-breadcrumb", () => {
...
     ).toBeTruthy;
   });
+
+  it("should call 'setFocus' when breadcrumb is focused", async () => {
+    const page = await newSpecPage({
+      components: [Breadcrumb],
+      html: `<ic-breadcrumb></ic-breadcrumb>`,
+    });
+
+    await page.rootInstance.focus;
```
Comment (ad9242) [inline packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts:129]: I don't think this line is doing anything in terms of testing the focus (in any of the focus tests). I think the actual execution of the setFocus function is coming from the `await page.rootInstance.setFocus` on the next line

**Review by ad9242 — CHANGES_REQUESTED** [review]

File: packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts (hunk 2 above)
Comment (GCHQ-Developer-847) [reply packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts:129]: I would imagine that in an 'expect' statement, once you have called setFocus(), you would check that document.activeElement is the correct HTML element (e.g. the 'ic-link' here) :)

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]

File: packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts (hunk 2 above)
Comment (GCHQ-Developer-847) [reply packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts:129]: I would imagine that in an 'expect' statement, once you have called setFocus(), you would check that document.activeElement is the correct HTML element (e.g. the 'ic-link' here) :)

File: packages/web-components/src/components/ic-card/ic-card.spec.ts (hunk 1 above)
Comment (GCHQ-Developer-530) [reply packages/web-components/src/components/ic-card/ic-card.spec.ts:198]: Ah yeah, meant to change it to say link

File: packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts (hunk 2 above)
Comment (GCHQ-Developer-530) [reply packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts:129]: This is how the tests are implemented for the existing setFocus methods, should I go through and update all of those?

Comment (GCHQ-Developer-530) [reply packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.spec.ts:129]: This is how the existing test for button's setFocus has been done. Not sure if the above is possible, it seems accessing the document.activeElement is only possible in e2e tests like `const activeElement = await page.evaluate(() => document.activeElement.id);` but testing there wouldn't make the coverage go up.

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**